src/utils.ts

```diff
@@ -177,8 +177,10 @@ export class FormattedMethodDescriptor implements MethodDescriptorProto {
 export function determineFieldJsonName(field: FieldDescriptorProto, options: Options): string {
   // By default jsonName is camelCased by the protocol compilier unless the user has
   // set a "json_name" option on this field.
-  if (field.jsonName.length > 0) {
+  const fromStringSnakeToCamel = typeof options.snakeToCamel === 'string' && options.snakeToCamel.includes('json');
+  const fromBooleanSnakeToCamel = options.snakeToCamel === true;
+  if (fromBooleanSnakeToCamel || fromStringSnakeToCamel) {
     return field.jsonName;
   }
-  return maybeSnakeToCamel(field.name, options);
+  return stringToSnakeCase(field.name);
```

Comment:
Ah yeah...I see what you're thinking here, but maybe just leave this out?

I'm thinking that if the user has set `snakeToCamel=false` that's not _exactly_ the same thing as "camelToSnake=true" i.e. requesting that we actively snake-ize fields in the JSON. So I'm tempted to just leave the field name as it is.